### PR TITLE
E4s oneapi force oneapi

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Force Oneapi

### CODE REVIEW

File: share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ spack:
     all:
       require:
       - "target=x86_64_v3"
-      prefer:
+      # prefer:
```
Comment (alalazo) [inline share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml]:
```suggestion
```

**Comment by alalazo** [comment]
Seems fine, I wonder why `%intel` got selected for a few packages - but `prefer` allows that at very low priority.

**Comment by kwryankrattiger** [comment]
> Seems fine, I wonder why `%intel` got selected for a few packages - but `prefer` allows that at very low priority.

It is really odd to me since oneapi is the `prefer`'d and the only other compiler mentioned at all in the stack is `gcc`. `intel` seems to get selected simply because it comes first in the `compilers.yaml` which it feels like that should be way less weighted compared to a listed preference.

**Comment by alalazo** [comment]
My guess is that there must be some other `any_of` requirement involved. If it's only compiler weight, then `prefer` is much more prioritized.

**Review by eugeneswalker — APPROVED** [review]

**Pull request merged**